plugins/modules/digital_ocean_domain.py

```diff
@@ -162,6 +161,19 @@ def core(module):
             if 'message' in domain:
                 module.fail_json(changed=False, msg=domain['message'])
             else:
+                # We're at the mercy of a backend process which we have no visibility into:
+                # https://developers.digitalocean.com/documentation/v2/#create-a-new-domain
+                #
+                # In particular: "Keep in mind that, upon creation, the zone_file field will
+                # have a value of null until a zone file is generated and propagated through
+                # an automatic process on the DigitalOcean servers."
+                #
+                # Arguably, it's nice to see the records versus null, so, we'll just try a
+                # few times before giving up and returning null.
+                for i in range(5):
+                    if do_manager.domain_record()['domain']['zone_file'] is not None:
```

Comment:
I'm not sure it's safe to index the domain record like this.

The `domain_record()` function just returns the response JSON without checking for errors. Isn't it possible that `domain` or `zone_file` won't exist?

Reply:
The `domain` key can be missing if the response code is not 200, but I am not sure if fixing this is in the scope of this PR. I would say that we should fix this (and other similar issues in the remaining data fetching methods) in a separate follow-up PR.

Reply:
I know this section of code is out of date, so this comment may be irrelevant now, but I'll have to disagree here. This is new code that has the potential to cause a traceback in the Python code as-is. For example, if the return is `None` or an empty dictionary you might see a `KeyError` or `TypeError`.

<details name="blah">
    <summary>Example REPL</summary>

```
>>> foo = {}
>>> foo['domain']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'domain'

>>> foo = None
>>> foo['domain']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not subscriptable
```

</details>

Reply:
(that said, I'm fine with resolving this in a separate PR as long as it's not in a release.)

Reply:
I see the code was changed as I suggested below, so there is no more traceback/regression potential here.

Sorry for the added noise on this PR. I'll resolve this conversation.